subsys/bluetooth/host/att.c

```diff
@@ -3088,9 +3105,42 @@ int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels)
 		return -ENOMEM;
 	}
 
-	return bt_l2cap_ecred_chan_connect(conn, chan, BT_EATT_PSM);
+	while (created < i) {
+		/* bt_l2cap_ecred_chan_connect() uses the first L2CAP_ECRED_CHAN_MAX_PER_REQ
+		 * elements of the array or until a null-terminator is reached.
+		 */
+		err = bt_l2cap_ecred_chan_connect(conn, &chan[created], BT_EATT_PSM);
+		if (err < 0) {
+			return err;
+		}
+
+		created += L2CAP_ECRED_CHAN_MAX_PER_REQ;
```

Comment:
Can `num_channels` be < `L2CAP_ECRED_CHAN_MAX_PER_REQ? 

If so, this is misleading (although wouldn't cause any issue as far as I can tell

Reply:
Yes, `num_channels` can be anything in the range [1, `CONFIG_BT_EATT_MAX`].

`bt_l2cap_ecred_chan_connect()` will request channels `L2CAP_ECRED_CHAN_MAX_PER_REQ` at a time. The last call will request fewer than that if `num_channels % L2CAP_ECRED_CHAN_MAX_PER_REQ != 0`. `created` will then be larger than `i` and exit the loop. It is then not equal to the number of channels requested so the name is somewhat misleading I guess.

Reply:
I suggest the following

1. Have `bt_l2cap_ecred_chan_connect` somehow return the number of channels attempted to connect and use that value to increment
2. Renamed the variable
3. change this line to `created = MIN(created + L2CAP_ECRED_CHAN_MAX_PER_REQ, i);